### PR TITLE
Adds Rspec failure messages.

### PR DESCRIPTION
**Pull request by vjustov**
This PR tackles #17.

Im not entirely comfortable with the messages per se, neither with how ActionHandlerMatchers turned out.

Let me know if i should change anything.

### CODE REVIEW

File: lib/resource_kit/testing/action_handler_matchers.rb [hunk 1]
```diff
@@ -21,16 +21,29 @@ def with(options, &block)
       def matches?(subject, &block)
         @handled_block ||= block
         action = subject.resources.find_action(self.action)
-        return false unless action
+        unless action
+          @failures = "expected :#{self.action} to be handled by the class."
```
Comment (bobbytables) [inline lib/resource_kit/testing/action_handler_matchers.rb]: Why not make the `attr_reader` be `failure_message` and just set this instance variable to that?

File: lib/resource_kit/testing/action_handler_matchers.rb [hunk 2]
```diff
@@ -21,16 +21,29 @@ def with(options, &block)
...
 
         true
       end
+
+      def failure_message
+        failures
+      end
+
+      private
+      attr_reader :failures
```
Comment (bobbytables) [inline lib/resource_kit/testing/action_handler_matchers.rb]: If you change this to be a public reader called `failure_message` then we don't need the middleman method.

File: spec/lib/resource_kit/testing/action_handler_matchers_spec.rb [hunk 3]
```diff
@@ -37,4 +37,32 @@
...
+        end
+
+        matcher.with(status: 200, body: '{"Hello": "World"}')
+        matcher.matches?(resource_class) { }
+
+        expect(matcher.failure_message).to eq('expected :all to be handled by the class.')
+      end
+    end
+  end
+end
```
Comment (bobbytables) [inline spec/lib/resource_kit/testing/action_handler_matchers_spec.rb]: These specs rock.

File: spec/lib/resource_kit/testing/have_action_matchers_spec.rb [hunk 4]
```diff
@@ -105,4 +105,53 @@
       end
     end
   end
-end
\ No newline at end of file
+
+  describe '#failure_message' do
+    let(:resource_class) {
+      Class.new(ResourceKit::Resource).tap { |resource|
```
Comment (bobbytables) [inline spec/lib/resource_kit/testing/have_action_matchers_spec.rb]: With `Class.new()` it actually yields a block in the context of the newly created class. So no need for `tap`.

File: lib/resource_kit/testing/action_handler_matchers.rb (hunk 1 above)
Comment (vjustov) [inline lib/resource_kit/testing/action_handler_matchers.rb]: I was trying to maintain homogeneity between the two matchers, but you are totally right.

**Comment by vjustov** [comment]
updated.

**Comment by vjustov** [comment]
Hello @bobbytables, I have made the requested changes.
Let me know if you have some others.

**Comment by bobbytables** [comment]
Love it. @phillbaker ?

**Comment by gotjosh** [comment]
:+1: this is really helpful!

**Comment by markmercedes** [comment]
:+1: great stuff

**Comment by phillbaker** [comment]
Thanks @vjustov & @bobbytables!

**Pull request merged**